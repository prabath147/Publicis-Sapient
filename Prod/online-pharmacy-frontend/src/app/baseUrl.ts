export const baseURL = "https://frk5iphcfk.execute-api.us-east-1.amazonaws.com/";
    // "http://localhost:8081";
    
