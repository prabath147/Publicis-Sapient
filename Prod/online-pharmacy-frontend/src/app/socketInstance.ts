
export const socketURL = "https://c8zpvs7fv4.execute-api.us-east-1.amazonaws.com/notify-ws"

// const socket = new SockJS(socketURL);
// export const stompClient = Stomp.over(socket);