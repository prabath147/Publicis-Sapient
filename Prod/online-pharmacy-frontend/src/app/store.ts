import { Action, configureStore, ThunkAction } from "@reduxjs/toolkit";
import StoreReducer from "../components/pages/Admin/storeList/StoreSlice";
import SubscriptionReducer from "../components/pages/AdminSubscriptionPack/SubscriptionvSlice";
import UserReducer from "../components/pages/Auth/Login/UserSlice";
import CartReducer from "../components/pages/Cart/CartSlice";
import ManagerReducer from "../components/pages/Manager/ManagerSlice";
import NotificationReducer from "../components/pages/Open/notificationList/NotificationSlice";
// import OrderSlice from "../components/pages/orderDetail/OrderSlice";
// import OrderListSlice from "../components/pages/orderList/OrderListSlice";

export const store = configureStore({
  reducer: {
    storeL: StoreReducer,
    user: UserReducer,
    notification: NotificationReducer,
    viewsubscription: SubscriptionReducer,
    // addsubscription: AddSubcriptionRe,
    cart: CartReducer,
    // order: OrderSlice,
    // orderL: OrderListSlice,
    manager: ManagerReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
