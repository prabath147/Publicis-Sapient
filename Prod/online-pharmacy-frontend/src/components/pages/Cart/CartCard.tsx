import { Card, Group, Text } from '@mantine/core';
import CartItemCard from '../item/CartItemCard';
import CartProductCard from '../item/CartProductCard';
import { ItemMiniDTO, ProductMiniDTO } from './models';

interface CartCardType {
    itemCart?: ItemMiniDTO[];
    productCart?: ProductMiniDTO[];
}
export default function CartCard({ itemCart, productCart }: CartCardType) {
    let cart: any[] = []

    if (itemCart !== undefined) {
        cart = itemCart
    } else if (productCart !== undefined) {
        cart = productCart
    }
    // console.log(cart)
    if (cart.length === 0) return <Text>No data Here!</Text>


    // calculate total price
    let temp_total = 0;
    for (const item of cart) {
        temp_total += item.price * getQantity(item)
    }
    const total = Math.round(temp_total * 100) / 100

    // to calculate quantity
    function getQantity(item: any) {
        if (item.itemQuantity !== undefined) {
            return item.itemQuantity
        } else {
            return item.quantity
        }
    }

    return (
        <>
            <Card>
                <Group position="apart" >
                    <Text>
                        No of Items: {cart.length}
                    </Text>

                    <Text>
                        Total Item Cost : â‚¹{total}
                    </Text>
                </Group>
                <br />
                <>

                    {
                        cart.map((item) => {

                            if (item.itemQuantity === undefined)
                                return <CartProductCard key={item.itemIdFk} item={item} />
                            else
                                return <CartItemCard key={item.productIdFk} item={item} />
                        })
                    }
                </>
            </Card>
        </ >
    )
}
