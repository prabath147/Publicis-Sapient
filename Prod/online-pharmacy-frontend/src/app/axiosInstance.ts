import axios, { AxiosRequestConfig } from "axios";
import LocalUserService from "../services/LocalUserService";
import TokenService from "../services/TokenService";
import { baseURL } from "./baseUrl";

const axiosInstance = axios.create({
  baseURL: baseURL,
  timeout: 5000,
  headers: {
    "Content-Type": "application/json",
    accept: "application/json",
  },
});

axiosInstance.interceptors.request.use(
  (config: AxiosRequestConfig) => {
    const token = TokenService.getLocalAccessToken();
    if (token && config.headers !== undefined) {
      config.headers["Authorization"] = "Bearer " + token; // for Spring Boot back-end
      // config.headers["x-access-token"] = token; // for Node.js Express back-end
      config.headers["ngrok-skip-browser-warning"] = "";
    }
    return config;
  },
  (error) => {
    return Promise.reject(error);
  }
);

axiosInstance.interceptors.response.use(
  (res) => {
    return res;
  },
  async (err) => {
    const originalConfig = err.config;
    if (originalConfig.url !== "api/auth/signin" && err.response) {
      // Access Token was expired
      if (err.response.status === 401 && !originalConfig._retry) {
        originalConfig._retry = true;
        try {
          const rs = await axiosInstance.post("api/auth/refresh-token", {
            refreshToken: TokenService.getLocalRefreshToken(),
          });
          const { accessToken } = rs.data;
          TokenService.updateLocalAccessToken(accessToken);
          return axiosInstance(originalConfig);
        } catch (_error) {
          TokenService.removeLocalAccessToken()
          TokenService.removeLocalRefreshToken()
          LocalUserService.removeLocalUser()
          // window.location.reload();
          return Promise.reject(_error);
        }
      }
    }
    return Promise.reject(err);
  }
);
export default axiosInstance;
