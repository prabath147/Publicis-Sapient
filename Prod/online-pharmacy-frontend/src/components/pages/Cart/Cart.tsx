import {
  Badge,
  Button,
  Center,
  Checkbox,
  Container,
  Grid,
  Group,
  Space,
  Text,
  Title,
  UnstyledButton
} from "@mantine/core";
import { showNotification } from "@mantine/notifications";
import { IconNotes } from "@tabler/icons";
import { FaMinus, FaPlus, FaShoppingCart, FaTrash } from "react-icons/fa";
import { Link } from "react-router-dom";
import { useAppDispatch, useAppSelector } from "../../../app/hooks";
import { getUserData } from "../Auth/Login/UserSlice";
import ItemLoadableCard from "../item/CartItemCard";
import {
  decreamentInCartAPI,
  EmptyCartAPI,
  increamentInCartAPI
} from "./CartAPI";
import {
  clearCart,
  decrementQuantity,
  getCartItemCount,
  getCartItems,
  getCartSelectedIds,
  getCartTotalAmount,
  incrementQuantity,
  removeItemFromCart,
  toggleSelect
} from "./CartSlice";

export default function Cart() {
  const cart = useAppSelector(getCartItems);
  const totalItem = useAppSelector(getCartItemCount);
  const totalAmount = useAppSelector(getCartTotalAmount);
  const cartSelected = useAppSelector(getCartSelectedIds);
  // console.log(cart)

  // console.log(cart);

  // const navigate = useNavigate();
  const uid = useAppSelector(getUserData).id;

  const dispatch = useAppDispatch();

  return (
    <>
      <Container>
        <Title>Cart</Title>

        <Group position="right">
          <Link to="/prescription">
            <IconNotes />
          </Link>

          <FaShoppingCart size={28} />
          <Text>{totalItem}</Text>
        </Group>

        {/* item total and price total */}
        <Group position="apart">
          <div>
            <Text>
              You have{" "}
              <Text color="blue" inline={true}>
                {totalItem}
              </Text>
              items in cart!
            </Text>
          </div>

          <div>
            <Text>Cart Total : â‚¹{totalAmount.toFixed(2)}</Text>
          </div>
        </Group>

        {/* cart actions */}
        <Center>
          <div style={{ borderRadius: 20 }}>
            <Link to={cartSelected.length === 0 ? "#" : "/orders/place"}>
              <Button disabled={cartSelected.length === 0} compact>
                Checkout
              </Button>
            </Link>

            <Link to={cartSelected.length === 0 ? "#" : "/optin/create"}>
              <Button
                compact
                variant="outline"
                disabled={cartSelected.length === 0}
              >
                Create Opt In
              </Button>
            </Link>

            <Button
              compact
              color="red"
              onClick={() => {
                EmptyCartAPI(uid)
                  .then((response) => {
                    dispatch(clearCart());
                    showNotification({
                      color: "green",
                      message: "Successfully emptied the cart",
                    });
                  })
                  .catch((err) => {
                    showNotification({
                      color: "red",
                      message: "oops, something went wrong",
                    });
                    console.log(err);
                  });
              }}
            >
              Empty Cart
            </Button>
          </div>
        </Center>

        <Space h="xl" />

        <Center>
          <div className="cart-items-container">
            {cart.map((curItem) => {
              return (
                <ItemLoadableCard
                  key={curItem.itemId}
                  item={curItem}
                  checkbox={
                    <Checkbox
                      checked={cartSelected.includes(curItem.itemIdFk)}
                      onChange={() => dispatch(toggleSelect(curItem.itemIdFk))}
                    />
                  }
                >
                  <>
                    <Grid.Col span={4}>
                      <UnstyledButton
                        onClick={() =>
                          // TODO add remove from cart link
                          decreamentInCartAPI(uid, {
                            itemIdFk: curItem.itemIdFk,
                            price: curItem.price,
                            itemQuantity: 1,
                            itemId: curItem.itemId,
                          })
                            .then((response) => {
                              dispatch(decrementQuantity(curItem.itemIdFk));
                            })
                            .catch((err) => {
                              showNotification({
                                color: "red",
                                message: "oops, something went wrong",
                              });
                              // console.log("lol");
                            })
                        }
                      >
                        <FaMinus size={9} />
                      </UnstyledButton>

                      <Badge size="md" style={{ margin: 4 }}>
                        {" "}
                        {curItem.itemQuantity}{" "}
                      </Badge>

                      <UnstyledButton
                        onClick={() => {
                          increamentInCartAPI(uid, {
                            itemIdFk: curItem.itemIdFk,
                            price: curItem.price,
                            itemQuantity: 1,
                            itemId: curItem.itemId,
                          })
                            .then((response) => {
                              dispatch(incrementQuantity(curItem.itemIdFk));
                            })
                            .catch((err) => {
                              console.log(err);
                            });
                        }}
                      >
                        <FaPlus size={9} />
                      </UnstyledButton>
                    </Grid.Col>

                    <Grid.Col span={4}>
                      <UnstyledButton
                        onClick={() =>
                          decreamentInCartAPI(uid, {
                            itemIdFk: curItem.itemIdFk,
                            price: curItem.price,
                            itemQuantity: curItem.itemQuantity,
                            itemId: curItem.itemId,
                          })
                            .then((response) => {
                              dispatch(
                                removeItemFromCart({
                                  itemIdFk: curItem.itemIdFk,
                                  price: curItem.price,
                                  itemQuantity: curItem.itemQuantity,
                                  itemId: curItem.itemId,
                                })
                              );
                            })
                            .catch((err) => {
                              console.log(err);
                            })
                        }
                      >
                        <FaTrash size={14} />
                      </UnstyledButton>
                    </Grid.Col>
                  </>
                </ItemLoadableCard>
              );
            })}
          </div>
        </Center>
      </Container>
    </>
  );
}
