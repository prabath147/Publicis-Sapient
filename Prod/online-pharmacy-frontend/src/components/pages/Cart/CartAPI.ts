import axiosInstance from "../../../app/axiosInstance";
import { ItemMiniDTO } from "./models";

export const getCartAPI = (id: number) => {
  return axiosInstance.get("order/cart/get-cart/" + id);
};

export const getItemDetailAPI = (item_fk_id: number) => {
  return axiosInstance.get("pharmacy/item/get-item/" + item_fk_id)
}

export const getProductDetailAPI = (productId: number) => {
  return axiosInstance.get("/pharmacy/product/get-product/" + productId)
}

export const addToCartAPI = (id: number, data: ItemMiniDTO) => {
  const purl = "/order/cart/add-to-cart/" + id;
  return axiosInstance.post(purl, data);
};

export const increamentInCartAPI = (id: number, data: ItemMiniDTO) => {
  const purl = "/order/cart/add-item/" + id;
  return axiosInstance.post(purl, data);
};

export const decreamentInCartAPI = (id: number, data: ItemMiniDTO) => {
  const purl = "/order/cart/subtract-item/" + id;
  return axiosInstance.post(purl, data);
};

export const EmptyCartAPI = (id: number) => {
  const purl = "/order/cart/empty-cart/" + id;
  return axiosInstance.delete(purl);
};

// export const removeFromCartAPI = (id: number, data: ItemMiniDTO) => {
//   const purl = "/order/cart/remove-from-cart/" + id;
//   return axiosInstance.put(purl, data);
// };