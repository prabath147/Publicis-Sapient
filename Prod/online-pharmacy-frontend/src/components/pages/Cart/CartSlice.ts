import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { RootState } from "../../../app/store";
import { CartSliceType, ItemMiniDTO } from "./models";

const initialState: CartSliceType = {
  items: [],
  selected: [],
  totalAmount: 0,
  totalItem: 0,
};

const CartSlice = createSlice({
  name: "cart",
  initialState,
  reducers: {
    loadCart: (state: CartSliceType, actions: PayloadAction<ItemMiniDTO[]>) => {
      console.log(actions.payload)
      state.items = actions.payload;
      // state.totalItem = actions.payload.quantity;
      // state.totalAmount = actions.payload.price;

      let totalPrice = 0;
      let totalQuantity = 0;

      for (const item of actions.payload) {
        totalPrice += item.itemQuantity * item.price;
        totalQuantity += item.itemQuantity
      }
      state.totalAmount = totalPrice;
      state.totalItem = totalQuantity;
    },


    addItemToCart: (state: CartSliceType, actions: PayloadAction<ItemMiniDTO>) => {
      state.items.push(actions.payload);
      state.totalItem += 1;
      state.totalAmount += actions.payload.price * actions.payload.itemQuantity;
    },

    removeItemFromCart: (
      state: CartSliceType,
      actions: PayloadAction<ItemMiniDTO>
    ) => {
      state.items = state.items.filter(
        (item) => item.itemIdFk !== actions.payload.itemIdFk
      );

      state.selected = state.selected.filter(
        (item) => item !== actions.payload.itemIdFk
      );

      state.totalItem -= 1;
      state.totalAmount -= actions.payload.price * actions.payload.itemQuantity;
    },

    incrementQuantity: (
      state: CartSliceType,
      actions: PayloadAction<number>
    ) => {
      state.items = state.items.map((curElem) => {
        if (curElem.itemIdFk === actions.payload) {
          state.totalAmount += curElem.price;
          // console.log('done');
          return { ...curElem, itemQuantity: curElem.itemQuantity + 1 };
        }
        return curElem;
      });
    },

    decrementQuantity: (
      state: CartSliceType,
      actions: PayloadAction<number>
    ) => {
      let total = 0;
      state.items = state.items
        .map((curElem) => {
          if (curElem.itemId === actions.payload) {
            state.totalAmount -= curElem.price;
            return { ...curElem, itemQuantity: curElem.itemQuantity - 1 };
          }
          return curElem;
        })
        .filter((curElem) => {
          if (curElem.itemQuantity !== 0) {
            total += 1;
            return true;
          }
          return false;
        });

      state.totalItem = total;
    },

    toggleSelect: (state: CartSliceType, actions: PayloadAction<number>) => {
      const index = state.selected.indexOf(actions.payload);

      if (index > -1) {
        state.selected = state.selected.filter((id) => id !== actions.payload);
      } else {
        state.selected.push(actions.payload);
      }
    },

    clearCart: (state: CartSliceType) => {
      // console.log("clear");

      state.items = [];
      state.selected = [];
      state.totalAmount = 0;
      state.totalItem = 0;
    },
  },
});

export const getCartItems = (state: RootState): ItemMiniDTO[] => {
  // return [{ itemId: 344, itemIdFk: 371157, price: 1, itemQuantity: 64 }]
  return state.cart.items
};

export const getSelectedCartItems = (state: RootState): ItemMiniDTO[] => {
  const selectedItems = state.cart.items.filter(
    (cartItem) => state.cart.selected.indexOf(cartItem.itemIdFk) !== -1
  );
  return selectedItems
};

export const getCartItemCount = (state: RootState): number => {
  return state.cart.totalItem;
};

export const getCartTotalAmount = (state: RootState): number => {
  // return Math.round(state.cart.totalAmount * 100) / 100;
  return state.cart.totalAmount
};

export const getCartSelectedIds = (state: RootState): number[] => {
  return state.cart.selected;
};

export const {
  loadCart,
  addItemToCart,
  removeItemFromCart,
  toggleSelect,
  incrementQuantity,
  decrementQuantity,
  clearCart,
} = CartSlice.actions;

export default CartSlice.reducer;
