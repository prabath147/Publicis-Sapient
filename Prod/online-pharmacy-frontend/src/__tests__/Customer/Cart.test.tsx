import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { useAppDispatch } from '../../app/hooks';
import { store } from '../../app/store';
import Cart from '../../components/pages/Cart/Cart';
import { clearCart, loadCart } from '../../components/pages/Cart/CartSlice';

jest.mock("../../components/pages/item/CartItemCard", () => () => (<>Component</>))

const itemList = [
    {
        itemIdFk: -1,
        itemQuantity: 1,
        price: 1,
        itemId: 1
    }
]

const MockCart = ({ items }) => {
    const dispatch = useAppDispatch()
    dispatch(clearCart())
    dispatch(loadCart(items))

    return <BrowserRouter>
        <Cart />
    </BrowserRouter>
}

describe('Cart', () => {

    it('component created with data', () => {
        const { asFragment } = render(
            <Provider store={store}  >
                <MockCart items={itemList} />
            </Provider>
        )
        expect(asFragment()).toMatchSnapshot()
    })

    it('component created with data and render', async () => {
        const { asFragment, getByTestId } = render(
            <Provider store={store}  >
                <MockCart items={[]} />
            </Provider>
        )

        // await waitForElementToBeRemoved(getByTestId(/Loading/i));

        expect(asFragment()).toMatchSnapshot()
    })


})
