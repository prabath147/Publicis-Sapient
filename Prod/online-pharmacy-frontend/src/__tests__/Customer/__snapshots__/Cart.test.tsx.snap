// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component created with data 1`] = `
<DocumentFragment>
  <div
    class="mantine-Container-root mantine-15pjuqq"
  >
    <h1
      class="mantine-Text-root mantine-Title-root mantine-dyfpg1"
    >
      Cart
    </h1>
    <div
      class="mantine-Group-root mantine-147cgkf"
    >
      <a
        class="mantine-gnzaph mantine-Group-child"
        href="/prescription"
      >
        <svg
          class="icon icon-tabler icon-tabler-notes"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <rect
            height="18"
            rx="2"
            width="14"
            x="5"
            y="3"
          />
          <line
            x1="9"
            x2="15"
            y1="7"
            y2="7"
          />
          <line
            x1="9"
            x2="15"
            y1="11"
            y2="11"
          />
          <line
            x1="9"
            x2="13"
            y1="15"
            y2="15"
          />
        </svg>
      </a>
      <svg
        class="mantine-gnzaph mantine-Group-child"
        fill="currentColor"
        height="28"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 576 512"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
        />
      </svg>
      <div
        class="mantine-Text-root mantine-Group-child mantine-1wsjsij"
      >
        1
      </div>
    </div>
    <div
      class="mantine-Group-root mantine-19jxmdp"
    >
      <div
        class="mantine-gnzaph mantine-Group-child"
      >
        <div
          class="mantine-Text-root mantine-1pbxw0k"
        >
          You have 
          <div
            class="mantine-Text-root mantine-9ts85k"
          >
            1
          </div>
          items in cart!
        </div>
      </div>
      <div
        class="mantine-gnzaph mantine-Group-child"
      >
        <div
          class="mantine-Text-root mantine-1pbxw0k"
        >
          Cart Total : ₹1.00
        </div>
      </div>
    </div>
    <div
      class="mantine-8od8ev"
    >
      <div
        style="border-radius: 20px;"
      >
        <a
          href="/"
        >
          <button
            class="mantine-UnstyledButton-root mantine-Button-root mantine-dteu3o"
            data-button="true"
            data-disabled="true"
            disabled=""
            type="button"
          >
            <div
              class="mantine-3xbgk5 mantine-Button-inner"
            >
              <span
                class="mantine-qo1k2 mantine-Button-label"
              >
                Checkout
              </span>
            </div>
          </button>
        </a>
        <a
          href="/"
        >
          <button
            class="mantine-UnstyledButton-root mantine-Button-root mantine-3gzr02"
            data-button="true"
            data-disabled="true"
            disabled=""
            type="button"
          >
            <div
              class="mantine-3xbgk5 mantine-Button-inner"
            >
              <span
                class="mantine-qo1k2 mantine-Button-label"
              >
                Create Opt In
              </span>
            </div>
          </button>
        </a>
        <button
          class="mantine-UnstyledButton-root mantine-Button-root mantine-1vy3keu"
          data-button="true"
          type="button"
        >
          <div
            class="mantine-3xbgk5 mantine-Button-inner"
          >
            <span
              class="mantine-qo1k2 mantine-Button-label"
            >
              Empty Cart
            </span>
          </div>
        </button>
      </div>
    </div>
    <div
      class="mantine-1eabysl"
    />
    <div
      class="mantine-8od8ev"
    >
      <div
        class="cart-items-container"
      >
        Component
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Cart component created with data and render 1`] = `
<DocumentFragment>
  <div
    class="mantine-Container-root mantine-15pjuqq"
  >
    <h1
      class="mantine-Text-root mantine-Title-root mantine-dyfpg1"
    >
      Cart
    </h1>
    <div
      class="mantine-Group-root mantine-147cgkf"
    >
      <a
        class="mantine-gnzaph mantine-Group-child"
        href="/prescription"
      >
        <svg
          class="icon icon-tabler icon-tabler-notes"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <rect
            height="18"
            rx="2"
            width="14"
            x="5"
            y="3"
          />
          <line
            x1="9"
            x2="15"
            y1="7"
            y2="7"
          />
          <line
            x1="9"
            x2="15"
            y1="11"
            y2="11"
          />
          <line
            x1="9"
            x2="13"
            y1="15"
            y2="15"
          />
        </svg>
      </a>
      <svg
        class="mantine-gnzaph mantine-Group-child"
        fill="currentColor"
        height="28"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 576 512"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
        />
      </svg>
      <div
        class="mantine-Text-root mantine-Group-child mantine-1wsjsij"
      >
        1
      </div>
    </div>
    <div
      class="mantine-Group-root mantine-19jxmdp"
    >
      <div
        class="mantine-gnzaph mantine-Group-child"
      >
        <div
          class="mantine-Text-root mantine-1pbxw0k"
        >
          You have 
          <div
            class="mantine-Text-root mantine-9ts85k"
          >
            1
          </div>
          items in cart!
        </div>
      </div>
      <div
        class="mantine-gnzaph mantine-Group-child"
      >
        <div
          class="mantine-Text-root mantine-1pbxw0k"
        >
          Cart Total : ₹1.00
        </div>
      </div>
    </div>
    <div
      class="mantine-8od8ev"
    >
      <div
        style="border-radius: 20px;"
      >
        <a
          href="/"
        >
          <button
            class="mantine-UnstyledButton-root mantine-Button-root mantine-dteu3o"
            data-button="true"
            data-disabled="true"
            disabled=""
            type="button"
          >
            <div
              class="mantine-3xbgk5 mantine-Button-inner"
            >
              <span
                class="mantine-qo1k2 mantine-Button-label"
              >
                Checkout
              </span>
            </div>
          </button>
        </a>
        <a
          href="/"
        >
          <button
            class="mantine-UnstyledButton-root mantine-Button-root mantine-3gzr02"
            data-button="true"
            data-disabled="true"
            disabled=""
            type="button"
          >
            <div
              class="mantine-3xbgk5 mantine-Button-inner"
            >
              <span
                class="mantine-qo1k2 mantine-Button-label"
              >
                Create Opt In
              </span>
            </div>
          </button>
        </a>
        <button
          class="mantine-UnstyledButton-root mantine-Button-root mantine-1vy3keu"
          data-button="true"
          type="button"
        >
          <div
            class="mantine-3xbgk5 mantine-Button-inner"
          >
            <span
              class="mantine-qo1k2 mantine-Button-label"
            >
              Empty Cart
            </span>
          </div>
        </button>
      </div>
    </div>
    <div
      class="mantine-1eabysl"
    />
    <div
      class="mantine-8od8ev"
    >
      <div
        class="cart-items-container"
      >
        Component
      </div>
    </div>
  </div>
</DocumentFragment>
`;
